app/models/user.rb

```diff
@@ -32,17 +37,41 @@ def all_friends
   end
 
   def approved_friend
-    all_friends.map(&:user_id)
+    me = all_friends.map(&:user_id)
+    you = all_friends.map(&:friend_id)
+    (me + you).uniq - [id]
+  end
+
+  def mutual_friend(myfriend)
+    me = approved_friend
+    you = myfriend.approved_friend
+    me & you
+  end
+
+  def mutual_friend_count(myfriend)
+    mutual_friend(myfriend).count
   end
```

Comment:
- [ ] This get just one side of the friendship. We need to add the inverse friends here too according to what we did before. Also it's possible to combine more query => `Post.where(user_id: [id] + friend_ids + inverse_friend_ids)`

Reply:
> I have fixed the errors pointed out in the last review. I tend to disagree on the last point, thus the and friend method this returns post of both the user and his friends. I intentionally place them on one line unlike before where I broke them doing.